### PR TITLE
feat!: make gov proposals with not sdk.Msgs invalid

### PR DESCRIPTION
**Pull request by evan-forbes**
## Description

this PR modifies the validate basic method of the MsgSubmitProposal type to invalidate proposals with no sdk.Msgs

closes https://github.com/celestiaorg/celestia-app/issues/1680

### CODE REVIEW

**Review by SweeXordious — APPROVED** [review]
Unrelated question:

What types of gov proposals are we currently supporting?

**Comment by evan-forbes** [comment]
> What types of gov proposals are we currently supporting?

all the standard ones except text proposals

File: x/gov/types/v1/msgs.go [hunk 1]
```diff
@@ -61,9 +61,9 @@ func (m MsgSubmitProposal) ValidateBasic() error {
 		return sdkerrors.Wrap(sdkerrors.ErrInvalidCoins, deposit.String())
 	}
 
-	// Check that either metadata or Msgs length is non nil.
-	if len(m.Messages) == 0 && len(m.Metadata) == 0 {
-		return sdkerrors.Wrap(types.ErrNoProposalMsgs, "either metadata or Msgs length must be non-nil")
+	// Check Msgs length is non nil.
```
Comment (rootulp) [inline x/gov/types/v1/msgs.go:64]: I see that this previously used the term "non nil" but I find that confusing b/c a non-nil `m.Messages` would return false here (i.e. if `len(m.Messages)=1).

```suggestion
	// Check if no Messages are proposed
```

File: x/gov/types/v1/msgs.go [hunk 2]
```diff
@@ -61,9 +61,9 @@ func (m MsgSubmitProposal) ValidateBasic() error {
...
+	if len(m.Messages) == 0 {
+		return sdkerrors.Wrap(types.ErrNoProposalMsgs, "Msgs length must be non-nil")
```
Comment (rootulp) [inline x/gov/types/v1/msgs.go:66]:
```suggestion
		return sdkerrors.Wrap(types.ErrNoProposalMsgs, "Msgs length must be non-zero")
```

**Review by rootulp — APPROVED** [review]

**Comment by rootulp** [comment]
[optional] a note in the README that describes this deviation from upstream cosmos-sdk

**Comment by evan-forbes** [comment]
@rootulp see spec update in https://github.com/celestiaorg/cosmos-sdk/pull/315/commits/67407a0047ef1693ae9dc672011bd9aa5c6b81be

**Pull request merged**